circuit/pallets/execution-delivery/src/lib.rs

```diff
@@ -883,91 +697,72 @@ impl<T: Config> Pallet<T> {
         contracts.extend(seen_contracts);
         contracts.extend(unseen_contracts);
 
-        let (current_block_no, block_zero) = (
+        let (_current_block_no, _block_zero) = (
             <frame_system::Pallet<T>>::block_number(),
             T::BlockNumber::zero(),
         );
 
-        let xtx_schedule = XtxSchedule::new_sequential_from_contracts(
-            contracts.clone(),
-            contract_ids.clone(),
-            composes.clone(),
-            action_descriptions.clone(),
-            None,
-            current_block_no.clone(),
-        )?;
+        Ok((contracts, contract_ids, action_descriptions))
+    }
 
-        let max_steps = contracts.len() as u32;
+    pub fn submit_xtx_execution(
+        contracts: Vec<RegistryContract<T::Hash, T::AccountId, BalanceOf<T>, T::BlockNumber>>,
+        requester: T::AccountId,
+        input: Vec<u8>,
+        value: BalanceOf<T>,
+        reward: BalanceOf<T>,
```

Comment:
Some notes here:
Maybe use a Vector of ContractIds and fetch the actual `RegistryContract` inside the function call
Also we need a way to enforce the `input` matches each of the provided contracts inputs? Or are we providing an input just to the first contract call?

Reply:
the input becomes a shared state as part of the VVM revamp. would leave it as is for now